.github/workflows/build_test_fmt.yml

```
@@ -0,0 +1,28 @@
+name: cargo-gccrs CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      master
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Check the coding style
+      run: |
+          cargo fmt -- --check
+    - name: Check for clippy warnings
+      env:
+        RUSTFLAGS: "-D warnings"
+      run: cargo clippy
+    - name: Build
+      env:
+        RUSTFLAGS: "-D warnings"
+      run: cargo build
+    - name: Run Unit Tests and Documentation examples
```

If you're not using the same `RUSTFLAGS` for test and build, you'll have to rebuild the project again for testing. I would either add this here, or just use this flag globally, so you won't forget to add it, if you add more steps in the future 

```suggestion
    - name: Run Unit Tests and Documentation examples
      env:
        RUSTFLAGS: "-D warnings"
```

---

I think it's good to set it up globally, at least for now! 1f8fb22

Thanks :D